File: hw_4_customview/src/main/java/com/gmail/elnora/fet/hw_4_customview/MainActivity.java
package com.gmail.elnora.fet.hw_4_customview;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;
import com.google.android.material.snackbar.Snackbar;
import java.util.Locale;

public class MainActivity extends AppCompatActivity {
    private static final String PREF_SAVE_KEY = "TOAST_SNACK";
    private static final String PREF_NAME = "PrefSettingActivity";
    private CustomView customView;
    private boolean settingChecked = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        settingChecked = loadSetting();

        customView = findViewById(R.id.customView);
        setOnTouchEventListener();
    }

    private void setOnTouchEventListener() {
        customView.setOnTouchEventListener(new CustomView.OnTouchEventListener() {
            @Override
            public void onTouchDown(int x, int y) {
                String message = String.format(Locale.getDefault(), "Нажаты координаты [%d;%d]", x, y);
                if (settingChecked != false) {
                    Snackbar.make(findViewById(R.id.parentLayout), message, Snackbar.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, message, Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == R.id.itemSettingShowMessage) {
            startSettingActivity();
        }
        return super.onOptionsItemSelected(item);
    }

    private void startSettingActivity() {
        Intent intent = new Intent(MainActivity.this, SettingActivity.class);
        startActivity(intent);
    }

    private boolean loadSetting() {
        SharedPreferences shared = getSharedPreferences(PREF_NAME, MODE_PRIVATE);
        return shared.getBoolean(PREF_SAVE_KEY, false);
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        settingChecked = loadSetting();
    }

}

File: hw_8_weather_app/src/main/java/com/gmail/elnora/fet/hw_8_weather_app/view/MainActivity.kt
package com.gmail.elnora.fet.hw_8_weather_app.view

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.gmail.elnora.fet.hw_8_weather_app.R
import com.gmail.elnora.fet.hw_8_weather_app.view.settings.SettingActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showMainFragmentWeather()
    }

    private fun showMainFragmentWeather() = showFragment(WeatherFragment.newInstance(), WeatherFragment.TAG)

    private fun showFragment(fragment: Fragment, tag: String) {
        supportFragmentManager.beginTransaction()
                .add(R.id.fragmentContainer, fragment, tag)
                .commit()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val itemId = item.itemId
        if (itemId == R.id.itemSettingShowMessage) {
            startSettingActivity()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun startSettingActivity() {
        val intent = Intent(this@MainActivity, SettingActivity::class.java)
        startActivity(intent)
    }

}
